tonic-web/src/layer.rs

```diff
@@ -4,32 +4,43 @@ use tower_layer::Layer;
 use tower_service::Service;
 
 /// Layer implementing the grpc-web protocol.
-#[derive(Debug, Clone)]
-pub struct GrpcWebLayer {
-    _priv: (),
+#[derive(Debug)]
+pub struct GrpcWebLayer<RespBody> {
+    _markers: std::marker::PhantomData<RespBody>,
 }
 
-impl GrpcWebLayer {
+impl<RespBody> Clone for GrpcWebLayer<RespBody> {
+    fn clone(&self) -> Self {
+        Self {
+            _markers: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<RespBody> GrpcWebLayer<RespBody> {
     /// Create a new grpc-web layer.
-    pub fn new() -> GrpcWebLayer {
-        Self { _priv: () }
+    pub fn new() -> Self {
+        Self {
+            _markers: std::marker::PhantomData,
+        }
     }
 }
 
-impl Default for GrpcWebLayer {
+impl<RespBody> Default for GrpcWebLayer<RespBody> {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl<S> Layer<S> for GrpcWebLayer
+impl<S, RespBody> Layer<S> for GrpcWebLayer<RespBody>
 where
-    S: Service<http::Request<hyper::Body>, Response = http::Response<BoxBody>>,
-    S: Send + 'static,
+    S: Service<http::Request<BoxBody>, Response = http::Response<RespBody>> + Send + 'static,
```

Comment:
This really broke my brain while working on it, but the BoxBody here is actually the Body that's forwarded from the GrpcWeb middleware to the upstream service. So it needs to be an actual type, because we don't just forward the body we get